### PR TITLE
Fix registry state lookup

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes [issue introduced](https://github.com/jupyter-widgets/ipywidgets/commit/ca91f899045ee11bd77c005754f7ffc76295026a#diff-522bb5e7695975cba0199c6a3d6df5be827035f4dc18ed6da22ac216b5615c77R397) on #3567

## Changes

- [x] use `_registry.get` instead of `register.get`
- [x] update official `actions/*` to latest versions to squash deprecation warnings
- [x] split up and upload failing results of `Test: JavaScript` job 

## Motivation

This is relatively difficult to demonstrate, as it requires a widget that creates a new client-side widget. For a reproducer see this [in-development notebook](https://wxyz--77.org.readthedocs.build/en/77/_static/lab/index.html?path=Examples/Filing.ipynb):
- run all cells
- try to add a new `.ipynb`

Screenshot from 2022-12-18 13-22-47

```py
File "/lib/python3.10/site-packages/ipykernel/comm.py", line 201, in comm_open
    f(comm, msg)
  File "/lib/python3.10/site-packages/ipywidgets/widgets/widget.py", line 421, in handle_comm_opened
    widget_class = register.get(state['_model_module'],
AttributeError: 'function' object has no attribute 'get'
```

- it's _possible_ to update the file with this patch, by downloading it and replacing `/lib/python3.10/site-packages/ipywidgets/widgets/widget.py`

## Future Work

Likely this should probably just get reviewed, and hopefully merged, as it's fairly clearly an improvement by inspection.

A test would emulate a client creating a new, pre-registered Widget.

However, a better long-term check would be applying `mypy`, `pyflakes` or the new-fangled (but very fast) `ruff` or some related tool which would do a better job at statically analyzing the code for catching these kinds of issues, but is probably out of scope for this one-line fix.

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Thanks @bollwyvl !
I've pushed a test that will avoid regressions.

And I agree, this could have been avoided. I'll put it on the agenda for next time we have a widget meeting.
(cc @jasongrout who was interested in this as well)

File: python/ipywidgets/ipywidgets/widgets/tests/test_widget.py [hunk 1]
```diff
@@ -74,3 +78,27 @@ def test_compatibility():
     caller_path = inspect.stack(context=0)[1].filename
     assert all(x.filename == caller_path for x in record)
     assert len(record) == 6
+
+
+def test_create_from_frontend():
```
Comment (bollwyvl) [inline python/ipywidgets/ipywidgets/widgets/tests/test_widget.py:83]: :tada:

**Comment by bollwyvl** [comment]
Beyond static analysis: it's probably worth pulling off the band-aid and applying `black`, `isort`, etc. Heck, I've even been using `ssort` recently on some projects... just more highly subjective things not to have to make a decision/talk about on PRs (after talking about the initial PR). I think just _doing_ it, with a single git history  ignore thing, rather than the slow `darken` process IPython has been following, is easier for most involved.

**Comment by maartenbreddels** [comment]
Yes, I think we all totally agree with you. I'd love to see it. What's your opinion on using pre-commit?

**Comment by maartenbreddels** [comment]
```
assert 'application/vnd.jupyter.widget-view+json' in mime_bundle, "widget should have have a view"
```
Also had this error locally...

**Comment by bollwyvl** [comment]
Fire, `pre-commit` is great... _in CI_. If it makes PRs to fix stuff it finds. I don't let it touch my machine, too many unmanaged environments. :rofl:

**Comment by maartenbreddels** [comment]
Hmm, some CI rot? Two seemingly unrelated workflows are failing. Is this something you touched in this PR @bollwyvl, or do you happen to know how to fix this?

**Comment by bollwyvl** [comment]
> seemingly unrelated workflows

Well, they both fail around playwright, of which I can't claim a lot of knowledge... I'm more concerned about the persist OoM on RTD, and the lack of meaningful output on the karma tests.

Without some of these things pinned (or at least archived) during a build, it's hard to say what has shifted. Perhaps a kicked run of `main` would shed some light on whether this is localized to this PR... I can roll back a bunch of changes, as I don't think there's much doubt about the critical line of the initial commit on this PR.

**Comment by bollwyvl** [comment]
I've slimmed this back down to just the fix and the tests, with [de23fa6](https://github.com/jupyter-widgets/ipywidgets/pull/3653/commits/de23fa6930fcee2a42c4cdd5542a121a3a27db60) being the last commit with the CI changes.

File: python/ipywidgets/ipywidgets/widgets/tests/test_send_state.py [hunk 2]
```diff
@@ -12,29 +12,20 @@
 # A widget with simple traits
 class SimpleWidget(Widget):
     a = Bool().tag(sync=True)
-    b = Tuple(Bool(), Bool(), Bool(), default_value=(False, False, False)).tag(sync=True)
+    b = Tuple(Bool(), Bool(), Bool(), default_value=(False, False, False)).tag(
```
Comment (bollwyvl) [inline python/ipywidgets/ipywidgets/widgets/tests/test_send_state.py:15]: can i roll these back, too?

**Comment by bollwyvl** [comment]
Even though the error is being discovered by playwright, i'll wager the issue lies with `jupyter_server 2.0.2`... haven't had a chance to look through the changes:

https://github.com/jupyter-server/jupyter_server/compare/v2.0.1...v2.0.2

**Comment by bollwyvl** [comment]
Maybe this:

https://github.com/jupyter-server/jupyter_server/compare/v2.0.1...v2.0.2#diff-72888b4a95043904b226c1aa5addc9687a1dd09706da5e59ed28be58ae5f99c9R2820

**Comment by bollwyvl** [comment]
https://github.com/jupyter-server/jupyter_server/issues/1143

**Comment by bollwyvl** [comment]
https://github.com/jupyter-server/jupyter_server/pull/1144

**Comment by bollwyvl** [comment]
That was merged... maybe a release forthcoming? Then we can get back to the _other_ CI problems here, i suppose...

**Comment by bollwyvl** [comment]
ok, that cleared up those two emergent fails. i'm still not sure if blocking this on fixing the docs/js CI fails is the right play, and would prefer to start a second PR to investigate further... though this work did lead to finding two upstream issues, so at least there's that.

**Comment by maartenbreddels** [comment]
Really appreciate the help here Nick!
I'm ok with merging this with docs failing (although that also needs fixing). If we can get the javascript failure sorted, that would make a lot of people happy (and I could do a new release).

**Comment by maartenbreddels** [comment]
Awesome work. I'm happy to merge this from this PR. I think I'll rebase/squash it into 2 or 3 commits somewhere tomorrow. Or, if you have your own opinion on how many commits there should be, feel free to do it yourself (I'm ok with force pushing)

**Comment by bollwyvl** [comment]
Squash it all!

**Comment by maartenbreddels** [comment]
I don't question authority 🙇  😄

**Comment by maartenbreddels** [comment]
This is released in ipywidgets 8.0.4

**Comment by maartenbreddels** [comment]
> I have basically given up trying to get `nbsphinx` to keep running.

Why is that? 

I haven't followed that development much, but I think you are saying myst-nb should replace nbsphinx?

I kind of like the pydata-sphinx-theme, looks clean. Maybe opening an issue on this so we can get some opinions in?

**Pull request merged**